src/org/labkey/test/util/TestDataGenerator.java

```diff
@@ -257,12 +257,17 @@ private Map<String, Object> generateRow()
         final Map<String, Object> newRow = new CaseInsensitiveHashMap<>();
         for (String columnName : _columns.keySet())
         {
-            if (_autoGeneratedFields.contains(columnName) || _dataSuppliers.containsKey(columnName))
+            if (!_dataSuppliers.containsKey(columnName))
+            {
+                _dataSuppliers.put(columnName, getDefaultDataSupplier(_columns.get(columnName).getRangeURI()));
+            }
```

Comment:
Sick Map trick 🛹 
```suggestion
            _dataSuppliers.computeIfAbsent(columnName, k -> getDefaultDataSupplier(_columns.get(columnName).getRangeURI()));
```

Reply:
I was excited to try this sick map trick, but unfortunately it didn't update the suppliers map; this meant that below the call to 

`_dataSuppliers.get(columnName).get();`

...would fail because _dataSuppliers.get(columnName) would be null ☹️